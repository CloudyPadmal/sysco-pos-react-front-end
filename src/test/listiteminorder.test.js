import React from 'react';
import { mountWrap } from './enzyme';

import ListItemInOrder from '../components/ListItemInOrder';

describe('ListItemInOrder', () => {

    it('displays item in an order', () => {
        const wrapper = mountWrap(<ListItemInOrder
            key={12123}
            ITEM={{ price: 45 }}
            NAME={`Item`}
            CLONEITEMQUANTITY={{}}
            ITEMQUANTITY={0}
            ADD_THIS_ITEM_TO_ITEMQUANTITY={() => { }}
            DELETE_THIS_ITEM={() => { }}
            INDECCREMENT_ITEM_COUNT={() => { }}
        />);
        expect(wrapper).toMatchSnapshot();
    });
});