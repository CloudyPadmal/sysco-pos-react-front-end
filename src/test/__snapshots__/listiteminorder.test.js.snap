// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemInOrder displays item in an order 1`] = `
<ListItemInOrder
  ADD_THIS_ITEM_TO_ITEMQUANTITY={[Function]}
  CLONEITEMQUANTITY={Object {}}
  DELETE_THIS_ITEM={[Function]}
  INDECCREMENT_ITEM_COUNT={[Function]}
  ITEM={
    Object {
      "price": 45,
    }
  }
  ITEMQUANTITY={0}
  NAME="Item"
>
  <li
    className="list-group-item d-flex justify-content-right"
    style={
      Object {
        "margin": "0px",
        "padding": "0px",
        "width": "100%",
      }
    }
  >
    <div
      className="row"
      style={
        Object {
          "margin": "0px",
          "width": "100%",
        }
      }
    >
      <div
        className="col-8 card"
      >
        <div
          className="card-body"
          style={
            Object {
              "padding": "5px 0px",
            }
          }
        >
          <h6
            className="card-title d-flex justify-content-right"
          />
          <p
            className="card-subtitle text-muted d-flex justify-content-right"
            style={
              Object {
                "textAlign": "initial",
              }
            }
          />
        </div>
      </div>
      <div
        className="col-2 card"
      >
        <div
          className="card-body d-flex align-items-center"
          style={
            Object {
              "padding": "5px 0px",
            }
          }
        >
          <ItemCounterInListItem
            ADD_THIS_ITEM_TO_ITEMQUANTITY={[Function]}
            COUNT={0}
            DELETE_THIS_ITEM={[Function]}
            INDECCREMENT_ITEM_COUNT={[Function]}
            MAX="undefined[object Object]"
            MIN={0}
            NAME="Item"
          >
            <div
              aria-label="Toolbar with button groups"
              className="btn-toolbar"
              role="toolbar"
            >
              <div
                aria-label="Item-Group"
                className="btn-group"
                role="group"
              >
                <button
                  className="btn btn-secondary"
                  onClick={[Function]}
                  type="button"
                >
                  +
                </button>
                <div
                  style={
                    Object {
                      "textAlign": "center",
                      "width": "3em",
                    }
                  }
                >
                  <span
                    className="input-group-text"
                    style={
                      Object {
                        "borderRadius": "0px",
                        "display": "inline-block",
                        "height": "100%",
                        "width": "100%",
                      }
                    }
                  >
                    0
                  </span>
                </div>
                <button
                  className="btn btn-secondary"
                  onClick={[Function]}
                  type="button"
                >
                  âˆ’
                </button>
              </div>
              <RenderDeleteButton
                COUNT={0}
                DELETE_THIS_ITEM={[Function]}
              />
            </div>
          </ItemCounterInListItem>
        </div>
      </div>
      <div
        className="col-2 card text-right"
      >
        <p
          className="card-text"
          style={
            Object {
              "margin": "5px 0 0 0",
            }
          }
        >
           Nos
        </p>
        <p
          className="card-link"
          style={
            Object {
              "margin": "0px",
            }
          }
        >
          @ 
          <b>
            Rs. 
            45.00
          </b>
        </p>
      </div>
    </div>
  </li>
</ListItemInOrder>
`;
